docs/source/hacking/documentation.rst

```diff
@@ -0,0 +1,70 @@
+Documenting Sensu Go Ansible collection
+=======================================
+
+Documentation for the Sensu Go Ansible collection consists of two distinct
+kinds: the general guides and the API documentation. The general guides live
+in the ``docs/source`` directory while the API documentation is part of the
+modules' source code.
+
+
+Authoring the documentation
+---------------------------
+
+Adding a new piece of general documentation is reasonably straightforward. We
+need to create a new file somewhere inside the docs/source directory and fill
+it with reStructuredText-formatted content.
+
+As a general rule, we use two levels of headings: a document title and as many
+section titles as needed. If you feel you need something more complex, try
+simplifying the documentation first. The end-users will thank you for this.
+
+First few lines of the document you are just reading look like this:
+
+.. literalinclude:: documentation.rst
+   :language: rst
+   :lines: 1-20
+
+Once we have the content ready, we need to inform Sphinx about it. We do this
+by adding our document to one of the ``toctree`` directives. We can find the
+important ones in the ``index.rst`` file.
+
+We follow the `upstream guides`_ when it comes to documenting the modules,
+with one exception: version_added field holds the Sensu Go collection version
+instead of Ansible version. Typical module starts with the next few lines:
+
+.. literalinclude:: ../../../plugins/modules/handler_set.py
+   :language: python
+   :lines: 1-30
+
+.. _upstream guides:
+   https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html
+
+
+Publishing the documentation
+----------------------------
+
+We use an ansible-doc-extractor tool to convert the embedded module
```

Comment:
```suggestion
We use the ``ansible-doc-extractor`` tool to convert the embedded module
```
I think it would be great to link to [xlab-si/ansible-doc-extractor](https://github.com/xlab-si/ansible-doc-extractor). Unless we want to avoid explaining why we're not using the official one? :)

In any case, I think we should point the readers either to our fork, or the upstream repo, or both, as they might not be familiar with `ansible-doc-extractor` at all, not even with the fact that this is Ansible's official tool for extracting collection docs.

Reply:
The current state of affairs is that there is no official tool for rendering module docs in Ansible. There is a helper script in Ansible repository with the sole purpose of rendering built-in documentation and is not meant to be used by 3rd party code. So our document extractor is not a fork per se since there is nothing to fork.

I added a link but avoided all the political motivation for our own specialized tool. That chapter in our development story is a nasty one ;)

Reply:
Okay, thanks for clarifying this. Totally fine with me to avoid anything but the link :)